core/src/main/java/net/sf/hajdbc/sql/CommonDataSource.java

```diff
@@ -87,7 +87,13 @@ public void close() throws SQLException
 	
 	public Z getProxy() throws SQLException
 	{
-		return this.createProxyFactory(this.getDatabaseCluster()).createProxy();
+		CommonDataSourceProxyFactory<Z, D> f = this.createProxyFactory(this.getDatabaseCluster());
+		try {
+			Z proxy = f.createProxy();
+			return proxy;
+		} finally {
+			this.getDatabaseCluster().removeListener(f);
+		}
```

Comment:
Actually, it's the proxy itself that needs cleanup following database deactivation.  Let me think about this some more.